privacyidea/api/lib/postpolicy.py

```diff
@@ -310,14 +310,14 @@ def no_detail_on_success(request, response):
     detailPol = policy_object.get_policies(action=ACTION.NODETAILSUCCESS,
                                            scope=SCOPE.AUTHZ,
                                            client=g.client_ip,
-                                           active=True,
-                                           audit_data=g.audit_object.audit_data)
+                                           active=True)
 
     if detailPol and content.get("result", {}).get("value"):
         # The policy was set, we need to strip the details, if the
         # authentication was successful. (value=true)
         del content["detail"]
         response.data = json.dumps(content)
+        g.audit_object.add_policy(list(detailPol.values())[0])
```

Comment:
Uff, now that I'm thinking of it: Theoretically, ``detailPol`` may return multiple policies that all set ``nodetailsuccess``. We will only add the first one to the audit log. But I think that's okay?

Reply:
This is wrong anyways! ``get_policies`` returns a list of policies. So using ``values()`` here is wrong.
We would simply do

    g.audit_object.add_policy(detailPol)